awx/main/tasks/receptor.py

```diff
@@ -405,10 +405,11 @@ def run(self):
         finally:
             # Make sure to always release the work unit if we established it
             if self.unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
-                try:
-                    receptor_ctl.simple_command(f"work release {self.unit_id}")
-                except Exception:
-                    logger.exception(f"Error releasing work unit {self.unit_id}.")
+                if settings.RECPETOR_KEEP_WORK_ON_ERROR and res and res.status == 'error':
```

Comment:
I'm pretty paranoid about this part of code, so do `getattr(res, 'status', 'error') == 'error'`, because after all, it could be `None`

Reply:
updated, i guess we also want the workunit to not release if res is None for some reason...